Add docs for preRenderedFetch param to render (#11256)
withastro/docs: src/content/docs/en/reference/adapter-reference.mdx

<p>

> **Type:** `{addCookieHeader?: boolean; clientAddress?: string; locals?: object; preRenderedFetch: (string: string) => Promise<Response>; routeData?: RouteData;}`

Reviewer: Looks like the type was updated in core but not here and `preRenderedFetch` is optional:
```suggestion
**Type:** `{addCookieHeader?: boolean; clientAddress?: string; locals?: object; preRenderedFetch?: (url: ErrorPagePath) => Promise<Response>; routeData?: RouteData;}`
```

Author: It's still optional in `RenderOptions`:
https://github.com/withastro/astro/pull/13403/files#diff-bfed2e2dbd4a6b01a9ef9d354a4b89031a3c070d5fa462e8232e59555ae3352aR78

Just not in `RenderErrorOptions` 🙃 

I'll fix the type though 🙏

> **Type:** `{addCookieHeader?: boolean; clientAddress?: string; locals?: object; preRenderedFetch?: (url: ErrorPagePath) => Promise<Response>; routeData?: RouteData;}`

Reviewer 2: ```suggestion
**Type:** `{addCookieHeader?: boolean; clientAddress?: string; locals?: object; prerenderedErrorPageFetch?: (url: ErrorPagePath) => Promise<Response>; routeData?: RouteData;}`
```

</p>

> The `app.render()` method accepts a mandatory `request` argument, and an optional `RenderOptions` object for [`addCookieHeader`](#addcookieheader), [`clientAddress`](#clientaddress), [`locals`](#locals), [`preRenderedFetch`](#prerenderedfetch), and [`routeData`](#routedata).

Reviewer 2: ```suggestion
The `app.render()` method accepts a mandatory `request` argument, and an optional `RenderOptions` object for [`addCookieHeader`](#addcookieheader), [`clientAddress`](#clientaddress), [`locals`](#locals), [`prerenderedErrorPageFetch`](#prerenderedErrorPageFetch), and [`routeData`](#routedata).
```

> The `app.render()` method accepts a mandatory `request` argument, and an optional `RenderOptions` object for [`addCookieHeader`](#addcookieheader), [`clientAddress`](#clientaddress), [`locals`](#locals), [`prerenderedErrorPageFetch`](#prerenderedErrorPageFetch), and [`routeData`](#routedata).

Reviewer 2: ```suggestion
The `app.render()` method accepts a mandatory `request` argument, and an optional `RenderOptions` object for [`addCookieHeader`](#addcookieheader), [`clientAddress`](#clientaddress), [`locals`](#locals), [`prerenderedErrorPageFetch`](#prerenderederrorpagefetch), and [`routeData`](#routedata).
```

}
```

###### `preRenderedFetch`

<p>

**Type:** `(url: ErrorPagePath) => Promise<Response>`<br />
**Default:** `fetch`<br />
<Since v="5.6.0" />
</p>

> In some environments, it would be useful to provide a custom implementation for fetching the prerendered error pages, for example if you don't have `fetch` available
> or cannot call the server from itself.
> 
> The example below reads 500.html and 404.html from disk instead of performing an HTTP call.

Reviewer 2: ```suggestion
A function that allows you to provide custom implementations for fetching prerendered error pages.

This is used to override the default `fetch()` behavior, for example when `fetch()` is unavailable or when you cannot call the server from itself.

The following example reads 500.html and 404.html from disk instead of performing an HTTP call:
```

}
```

> ###### `preRenderedFetch`

Reviewer 2: ```suggestion
###### `prerenderedErrorPageFetch`
```

<p>

> **Type:** `preRenderedFetch`<br />

Reviewer: We should use the exact type here:
```suggestion
**Type:** `(url: ErrorPagePath) => Promise<Response>`<br />
```

> **Default:** `fetch`
> </p>
> 
> <p><Since v="5.6.0" /></p>

Reviewer: `Since` should be [in the same block](https://contribute.docs.astro.build/reference/api-references/) as Type/Default:
```suggestion
**Default:** `fetch`<br />
<Since v="5.6.0" />
</p>
```

A function that allows you to provide custom implementations for fetching prerendered error pages.

> This is used to override the default `fetch()` behavior, for example when `fetch()` is unavailable or when you cannot call the server from itself.

Reviewer 3: comma nit
```suggestion
This is used to override the default `fetch()` behavior, for example, when `fetch()` is unavailable or when you cannot call the server from itself.
```

> The example bellow reads 500.html and 404.html from disk instead of performing an HTTP call.

Reviewer: typo:
```suggestion
The example below reads 500.html and 404.html from disk instead of performing an HTTP call.
```

> ```js "preRenderedFetch"
> return app.render(request, {
>   preRenderedFetch: async (url: string): Promise<Response> => {

Reviewer 2: ```suggestion
```js "prerenderedErrorPageFetch"
return app.render(request, {
  prerenderedErrorPageFetch: async (url: string): Promise<Response> => {
```
